# Menentukan Pertanyaan Bisnis

1. Skill apa yang paling umum dibutuhkan dalam tiap job function berdasarkan analisis job description?
   - Memberikan insight bagi pencari kerja untuk mengetahui skill yang meningkatkan peluang mendapat pekerjaan yang sesuai.
   - Membantu pelamar mengetahui posisi mereka dibanding pasar tenaga kerja.
2. Bagaimana karakteristik lowongan berdasarkan education level, career level dan experience level di berbagai industri?
   - Membantu kandidat memahami industri mana yang banyak membuka peluang di level karir tertentu.
   - Berguna untuk perusahaan dalam kompetisi talent acquisition dan strategi hiring.

# Import Semua Packages/Library yang Digunakan

In [ ]:
import pandas as pd

# Data Wrangling

## Gathering Data

In [2]:
df = pd.read_csv("../../data/all.csv", delimiter="|")
df.head()

id                                       job_title       location  \
0   1  Facility Maintenance & Smart Warehouse Manager        Bandung   
1   2                     Procurement Department Head   Jakarta Raya   
2   3                                     SALES ADMIN  Jakarta Barat   
3   4    City Operation Lead Shopee Express (Cirebon)        Cirebon   
4   5                            Japanese Interpreter         Bekasi   

  salary_currency                              career_level experience_level  \
0             IDR                   Manajer/Asisten Manajer          5 tahun   
1             IDR                   Manajer/Asisten Manajer          5 tahun   
2             IDR                    Supervisor/Koordinator          4 tahun   
3             IDR                    Supervisor/Koordinator          5 tahun   
4             IDR  Pegawai (non-manajemen & non-supervisor)          2 tahun   

                                     education_level employment_type  \
0  Sertifikat Professional, D3 (Diploma), D4 (Dip...     Penuh Waktu   
1  Sarjana (S1), Diploma Pascasarjana, Gelar Prof...     Penuh Waktu   
2                                       Sarjana (S1)     Penuh Waktu   
3  Sarjana (S1), Diploma Pascasarjana, Gelar Prof...     Penuh Waktu   
4  Sertifikat Professional, D3 (Diploma), D4 (Dip...     Penuh Waktu   

                              job_function  \
0                  Manufaktur,Pemeliharaan   
1  Manufaktur,Pembelian/Manajemen Material   
2    Penjualan / Pemasaran,Penjualan Ritel   
3        Pelayanan,Logistik/Rantai Pasokan   
4                  Lainnya,Jurnalis/Editor   

                                        job_benefits company_process_time  \
0                                                NaN                  NaN   
1                                                NaN              25 days   
2  Waktu regular, Senin - Jumat;Bisnis (contoh: K...              30 days   
3  Tip;Waktu regular, Senin - Jumat;Kasual (conto...              21 days   
4                                                NaN              23 days   

          company_size         company_industry  \
0                  NaN                      NaN   
1     51 - 200 pekerja  Manajemen/Konsulting HR   
2     51 - 200 pekerja            Umum & Grosir   
3  2001 - 5000 pekerja       Retail/Merchandise   
4    201 - 500 pekerja  Manajemen/Konsulting HR   

                                     job_description  salary  
0  Deskripsi PekerjaanRequirements :D3/SI from re...     NaN  
1  Job Role: 1. Responsible for material availabi...     NaN  
2  Internal Sales & AdminJob Description :We are ...     NaN  
3  Job Description:Responsible for HSE implementa...     NaN  
4  Overview: Our clients is manufacture for autom...     NaN

Merujuk dari dokumentasi datasetnya, berikut ini merupakan penjelasan untuk setiap kolom yang ada:

- job_title = nama pekerjaan
- location = lokasi perusahaan
- salary_currency = mata uang yang digunakan untuk gaji
- career_level = level karir
- experience_level = level pengalaman yang diharuskan untuk pelamar
- education_level = strata pendidikan yang diharuskan untuk pelamar
- employment_type = full-time, part-time, atau internship
- job_function = kategori pekerjaan
- job_benefits = benefit yang diberikan oleh perusahaan
- company_process_time = rata-rata waktu respon perusahaan
- company_size = jumlah karyawan
- company_industry = sektor industri dari perusahaan
- job_description = deskripsi pekerjaan
- salary = gaji yang ditawarkan perusahaan

## Assessing Data

#### Menilai data df

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34746 entries, 0 to 34745
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    34746 non-null  int64  
 1   job_title             34746 non-null  object 
 2   location              34746 non-null  object 
 3   salary_currency       34742 non-null  object 
 4   career_level          34746 non-null  object 
 5   experience_level      30205 non-null  object 
 6   education_level       34746 non-null  object 
 7   employment_type       33402 non-null  object 
 8   job_function          34746 non-null  object 
 9   job_benefits          27330 non-null  object 
 10  company_process_time  24555 non-null  object 
 11  company_size          29103 non-null  object 
 12  company_industry      33132 non-null  object 
 13  job_description       34745 non-null  object 
 14  salary                9352 non-null   float64
dtypes: float64(1), int6

Dataframe `df` memiliki 34.746 entri dan memiliki 14 kolom dengan kolom `salary_currency`, `experience_level`, `employment_type`, `job_benefits`, `company_process_time`, `company_size`, `company_industry`, `job_description`, `salary` memiliki data yang hilang. Sebagian besar kolom bertipe object dan hanya kolom `id` yang bertipe `int64` serta kolom `salary` bertipe `float64`.

#### Pemeriksaan parameter statistik

Selanjutnya, dilakukan pemeriksaan ringkasan parameter statistik seperti `count`, `mean`, `std`, `min`, dan `max`untuk kolom `id` dan `salary` yang bertipe numerik.

In [8]:
df.describe()

id        salary
count  34746.000000  9.352000e+03
mean   17373.500000  7.358879e+06
std    10030.450563  2.148128e+07
min        1.000000  1.000000e+01
25%     8687.250000  4.250000e+06
50%    17373.500000  5.500000e+06
75%    26059.750000  8.000000e+06
max    34746.000000  2.000000e+09

Kolom `id` memiliki `count` sebesar 34.746 yang nilainya sama dengan jumlah baris, artinya tidak ada id yang hilang.

Kolom `salary` memiliki `count` sebesar 9.352 yang artinya hanya 27% dari keseluruhan data yang memiliki gaji, hal ini mengindikasikan banyak missing values. Untuk `mean` dari kolom `salary` sendiri sekitar 7.3 juta. Standar deviasi dari kolom `salary` sebesar 21.48 juta yang mana ini sangat besar menandakan ada outlier besar di dalam data. Nilai minimum gaji adalah 10, kemungkinan karena error input. Adapun untuk gaji berada di rentang 4-8 juta yang mana hal ini terlihat konsisten. Terakhir ada nilai `max` sebesar 2 miliar yang menunjukkan adanya outlier ekstrem.

## Data Cleaning

#### Pemeriksaan duplikasi

Langkah berikut ini untuk melihat jumlah data duplikat yang terdapat di dataframe

In [10]:
print("Jumlah data duplikat dari df:", df.duplicated().sum())

Jumlah data duplikat dari df: 0


Tidak ditemukan adanya duplikat di dataframe `df`.

#### Menangani missing values

Langkah berikut ini untuk melihat jumlah missing values yang terdapat di dataframe

In [11]:
df_clean = df.copy()
df_clean.isnull().sum()

id                          0
job_title                   0
location                    0
salary_currency             4
career_level                0
experience_level         4541
education_level             0
employment_type          1344
job_function                0
job_benefits             7416
company_process_time    10191
company_size             5643
company_industry         1614
job_description             1
salary                  25394
dtype: int64

Selanjutnya dilakukan penanganan untuk kolom yang mempunyai missing values.

In [12]:
df_clean = df_clean.dropna()
df_clean.isnull().sum()

id                      0
job_title               0
location                0
salary_currency         0
career_level            0
experience_level        0
education_level         0
employment_type         0
job_function            0
job_benefits            0
company_process_time    0
company_size            0
company_industry        0
job_description         0
salary                  0
dtype: int64

Mengecek informasi umum dataframe setelah dilakukan penanganan missing values

In [15]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5044 entries, 9 to 34745
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    5044 non-null   int64  
 1   job_title             5044 non-null   object 
 2   location              5044 non-null   object 
 3   salary_currency       5044 non-null   object 
 4   career_level          5044 non-null   object 
 5   experience_level      5044 non-null   object 
 6   education_level       5044 non-null   object 
 7   employment_type       5044 non-null   object 
 8   job_function          5044 non-null   object 
 9   job_benefits          5044 non-null   object 
 10  company_process_time  5044 non-null   object 
 11  company_size          5044 non-null   object 
 12  company_industry      5044 non-null   object 
 13  job_description       5044 non-null   object 
 14  salary                5044 non-null   float64
dtypes: float64(1), int64(1), 